sound/soc/sof/intel/hda-ipc.c

```diff
@@ -267,6 +274,14 @@ void hda_ipc_msg_data(struct snd_sof_dev *sdev,
 					  struct sof_intel_hda_stream,
 					  hda_stream.hstream);
 
+		if (sz + hda_stream->stream.posn_offset > sdev->stream_box.size +
+		    sdev->stream_box.offset) {
```

Comment:
@plbossart I think it compares the end of the copying area with the end of the stream box. On the RHS `sdev->stream_box.offset + sdev->stream_box.size` seems to be the end of the stream box, on the LHS `hda_stream->stream.posn_offset + sz` is the end of the area, where the data is read from. I'd swap the order of the summands though to make it `offset + size` instead of `size + offset` but it should be commutative :-)